### PR TITLE
br: use atomic.Pointer instead of atomic.Value

### PR DESCRIPTION
**Pull request by yoshikipom**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/44736

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > Current Tests have already covered, so no need to add more tests.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/streamhelper/collector.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ type storeCollector struct {
 
 	input chan RegionWithLeader
 	// the oneshot error reporter.
-	err *atomic.Value
+	err *atomic.Pointer[error]
```
Comment (lance6716) [inline br/pkg/streamhelper/collector.go:41]: You can try `atomic.Error` from go.uber.org/atomic

File: br/pkg/storage/memstore.go [hunk 2]
```diff
@@ -135,7 +125,7 @@ func (s *MemStorage) ReadFile(ctx context.Context, name string) ([]byte, error)
 	if !ok {
 		return nil, errors.Errorf("cannot find the file: %s", name)
 	}
-	fileData := theFile.GetData()
+	fileData := *theFile.Data.Load()
```
Comment (lance6716) [inline br/pkg/storage/memstore.go:128]: it's safer to check the pointer is nil.

**Review by lance6716 — COMMENTED** [review]
Thanks for your contribution, and please check CI link to update the test cases

**Comment by lance6716** [comment]
don't worry, ci seems failed by other problems. I'll check it tomorrow

**Comment by yoshikipom** [comment]
@lance6716  Thank you for your support!

**Comment by lance6716** [comment]
Hi @yoshikipom please merge master branch, integration tests should be okay now

**Review by lance6716 — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**